src/BloomBrowserUI/bookEdit/js/textOverPicture.ts

```diff
@@ -31,16 +40,237 @@ class TextOverPictureManager {
+        // things in for editing.
+        // It causes EditingModel.RethinkPageAndReloadIt() to get run... which eventually causes
+        // makeTextOverPictureBoxDraggableClickableAndResizable to get called by bloomEditing.ts.
+        BloomApi.postThatMightNavigate("common/saveChangesAndRethinkPageEvent");
+    }
+
+    // Adds a new text-over-picture element as a child of the specified {parentElement}
+    //    (It is a child in the sense that the Comical library will recognize it as a child)
+    // {offsetX}/{offsetY} is the offset in position from the parent to the child elements
+    //    (i.e., offsetX = child.left - parent.left)
+    //    (remember that positive values of Y are further to the bottom)
+    // Note: After reloading the page, you can't have any of your other code execute safely
+    public addChildTOPBoxAndReloadPage(
+        parentElement: HTMLElement,
+        offsetX: number,
+        offsetY: number
+    ): void {
+        const parentBoundingRect = parentElement.getBoundingClientRect();
+        let newX = parentBoundingRect.left + offsetX;
+        let newY = parentBoundingRect.top + offsetY;
+
+        // // Ensure newX and newY is within the bounds of the container.
+        const container = parentElement.closest(".bloom-imageContainer");
+        if (!container) {
+            toastr.warning("Failed to create child element.");
+            return;
+        }
+        const containerBoundingRect = container.getBoundingClientRect();
+
+        const bufferPixels = 15;
+        if (newX < containerBoundingRect.left) {
+            newX = containerBoundingRect.left + bufferPixels;
+        } else if (
+            newX + parentElement.clientWidth >
+            containerBoundingRect.right
+        ) {
+            // ENHANCE: parentElement.clientWidth is just an estimate of the size of the child's width.
+            //          It would be better if we could actually plug in the real value of the child's width
+            newX = containerBoundingRect.right - parentElement.clientWidth;
+        }
+
+        if (newY < containerBoundingRect.top) {
+            newY = containerBoundingRect.top + bufferPixels;
+        } else if (
+            newY + parentElement.clientHeight >
+            containerBoundingRect.bottom
+        ) {
+            // ENHANCE: parentElement.clientHeight is just an estimate of the size of the child's height.
+            //          It would be better if we could actually plug in the real value of the child's height
+            newY = containerBoundingRect.bottom - parentElement.clientHeight;
+        }
+
+        const childElement = this.addFloatingTOPBox(newX, newY);
+        if (!childElement) {
+            toastr.info("Failed to place a new child callout.");
+            return;
+        }
+
+        Comical.initializeChild(childElement, parentElement);
+
+        // Need to reload the page to get it editable/draggable/etc.
```

Comment:
Add to comment, "and to get the Comical bubbles consistent with the new bubble specs"